### PR TITLE
CIV-9475 respondent agreement screen added

### PR DESCRIPTION
**Pull request by david-jeronimo**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link ###
https://tools.hmcts.net/jira/browse/CIV-9475

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/routes/urls.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export const BASE_ELIGIBILITY_URL = '/eligibility';
 export const BASE_FIRST_CONTACT_URL = '/first-contact';
 export const BASE_GENERAL_APPLICATION_URL = `${BASE_CASE_URL}/general-application`;
 export const APPLICATION_TYPE_URL = `${BASE_GENERAL_APPLICATION_URL}/application-type`;
+export const RESPONDENT_AGREEMENT_URL = `${BASE_GENERAL_APPLICATION_URL}/respondent-agreement`;
```
Comment (portizdev) [inline src/main/routes/urls.ts:7]: Pls change the URL. We are using this as the basis for the respondent's response: 
`export const BASE_GENERAL_APPLICATION_RESPONSE_URL = `${BASE_CASE_RESPONSE_URL}/general-application`;`
https://hmcts-reform.slack.com/archives/C02J4Q3QKLZ/p1715937364168269

File: src/main/services/features/generalApplication/generalApplicationService.ts [hunk 2]
```diff
@@ -48,6 +49,18 @@ export const saveInformOtherParties = async (redisKey: string, informOtherPartie
   }
 };
 
+export const saveRespondentAgreement = async (redisKey: string, respondentAgreement: RespondentAgreement): Promise<void> => {
+  try {
+    const claim = await getCaseDataFromStore(redisKey);
+    claim.generalApplication = Object.assign(new GeneralApplication(), claim.generalApplication);
```
Comment (portizdev) [inline src/main/services/features/generalApplication/generalApplicationService.ts:55]: For the response of the application, in order not to mix the fields of the creation of the application with the response.
Create a separate class (even inside generalApplication) to include all the fields from the response to the application.

**Review by portizdev — CHANGES_REQUESTED** [review]

**Review by portizdev — APPROVED** [review]

File: src/test/a11y/ignored-urls.ts [hunk 3]
```diff
@@ -58,7 +58,9 @@ export const IGNORED_URLS = [
   urls.CANCEL_URL,
   urls.VIEW_MEDIATION_DOCUMENTS,
   urls.BASE_GENERAL_APPLICATION_URL,
+  urls.BASE_GENERAL_APPLICATION_RESPONSE_URL,
   urls.DASHBOARD_CLAIMANT_URL,
+  urls.GA_RESPONDENT_AGREEMENT_URL,
```
Comment (laarmada) [inline src/test/a11y/ignored-urls.ts:63]: avoid ignore the URL if the accessibility test fail, please add the html under the correct folder:
image

File: src/main/views/features/generalApplication/respondent-agreement.njk [hunk 4]
```diff
@@ -0,0 +1,54 @@
+{% extends "claim-details-tpl-dashboard.njk" %}
```
Comment (laarmada) [inline src/main/views/features/generalApplication/respondent-agreement.njk:1]: please refactor the njk as the follow PR: https://github.com/hmcts/civil-citizen-ui/pull/3828

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/main/common/models/generalApplication/response/response.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { RespondentAgreement } from './respondentAgreement';
+
+export type Response = {
```
Comment (laarmada) [inline src/main/common/models/generalApplication/response/response.ts:3]: is better create a class instead of type

**Review by laarmada — CHANGES_REQUESTED** [review]

File: src/main/common/models/generalApplication/response/response.ts (hunk 5 above)
Comment (david-jeronimo) [reply src/main/common/models/generalApplication/response/response.ts]: done

File: src/main/views/features/generalApplication/respondent-agreement.njk (hunk 4 above)
Comment (david-jeronimo) [reply src/main/views/features/generalApplication/respondent-agreement.njk:1]: Done

File: src/test/a11y/ignored-urls.ts (hunk 3 above)
Comment (david-jeronimo) [reply src/test/a11y/ignored-urls.ts]: Done. Thanks for the info!

**Review by portizdev — DISMISSED** [review]

**Review by jarekPierchala — DISMISSED** [review]

**Review by neeta-hmcts — DISMISSED** [review]

**Pull request merged**